milli/src/update/index_documents/extract/extract_word_pair_proximity_docids.rs

```diff
@@ -23,8 +24,20 @@ use crate::{DocumentId, Result};
 pub fn extract_word_pair_proximity_docids<R: io::Read + io::Seek>(
     docid_word_positions: grenad::Reader<R>,
     indexer: GrenadParameters,
+    settings_diff: &InnerIndexSettingsDiff,
 ) -> Result<grenad::Reader<BufReader<File>>> {
     puffin::profile_function!();
+    let any_deletion = settings_diff.old.proximity_precision == ProximityPrecision::ByWord;
+    let any_addition = settings_diff.new.proximity_precision == ProximityPrecision::ByWord;
+
+    // early return if the data shouldn't be deleted nor created.
+    if !any_deletion && !any_addition {
+        return tempfile::tempfile()
+            .map_err(Into::into)
+            .map(BufReader::new)
+            .and_then(grenad::Reader::new)
+            .map_err(Into::into);
+    }
```

Comment:
If both versions of the settings have `ProximityPrecision::ByAttribute`, it means that the database is already empty and must remain like this, so the process is skipped